minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/ClusterServerConfiguration.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.miniclusterImpl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.accumulo.core.lock.ServiceLockData;
+
+public class ClusterServerConfiguration {
```

Comment:
This class will need to move to the public API or the public API will need something similar.  

Also thinking it would be nice to make the class immutable, this could be done with a  builder.


```java
class ClusterServerConfiguration {
   static class Builder {
          private Map<String,Integer> compactors = COMPACTORS_DEFAULT;
          private Map<String,Integer> sservers = SSERVERS_DEFAULT;
          private Map<String,Integer> tservers = TSERVERS_DEFAULT;

          Builder setScanServers(Map<String,Integer> scanServerGroups, boolean appendDefautls){
            if(!appendDefaults){
               sservers = Map.copyOf(scanServerGroups);
             } else {
               var copy = new HashMap(scanServerGroups);
               copy.putAll(SSERVERS_DEFAULT);
               // make it immutable
               sservers = Map.copyOf(copy);
             }
             return this;
          }

          Builder setTabletServers(Map<String,Integer> tabletServerGroups, boolean appendDefautls){
             //TODO
          }

           Builder setCompactors(Map<String,Integer> compactorGroups, boolean appendDefautls){
            // TODO
          }

         ClusterServerConfiguration build(){
            // this could be a private constructor
            return new ClusterServerConfiguration(tservers, sservers, compactors);
        }
   }

  static Builder build()  { return new Builder();}
}
```

Reply:
I can move to public API, but it's not immutable. In the tests we are starting and stopping services. Currently you modify the configuration and call start (if you want more compactors for example). I don't want to swap out the config for a brand new one at runtime.

Reply:
I was thinking when a new config is set it reconciles that with what is currently running and starts/stops processes.  Not sure what that would look like in practice.

We could also have an internal non public API version that is mutable.  I think making the API version immutable is nice as minis public API does not allow starting and stopping servers after the cluster starts, that seems to be something internal that we use for testing.

Reply:
Take a look at ExternalCompation1_IT in this PR for how I'm using it currently.

Reply:
I'll create a follow-on to evaluate putting ClusterServerConfiguration  in the public API, I want to get this merged so I can fold it into #3496 and keep moving forward.